lib/dotcom_web/live/trip_planner.ex

```diff
@@ -204,9 +204,25 @@ defmodule DotcomWeb.Live.TripPlanner do
   def handle_event("select_itinerary", %{"index" => index}, socket) do
     index = String.to_integer(index)
 
+    new_map = %{
```

Comment:
There is a fundamental issue here, which is that `lines` and `points` have to be re-computed at various different points, and forgetting to do that somewhere introduces bugs like this one. (It also creates a bunch of duplicative code for re-calculating `lines` and `points` any time they ought to change)

I think the Right™ solution is to calculate `lines` and `points` (and a few other things, like `pins`) at the top of `render`, based on the result set and user interactions, and not try to have this one-to-one mapping between components-on-screen and top-level state fields.

I'd rather actually tackle that refactor as a follow-up, since fixing this bug in this way was pretty easy, and because trying to refactor the state right now risks conflicting with other ongoing trip planner work.

Reply:
⭐ Thanks for this! I agree with your take on what an improvement on this would be.